mlflow/pytorch/__init__.py

```diff
@@ -377,31 +380,48 @@ def save_model(
     # Save pytorch model
     model_path = os.path.join(model_data_path, _SERIALIZED_TORCH_MODEL_FILE_NAME)
 
+    torchserve_artifacts_config = {}
+
     if requirements_file:
         if not isinstance(requirements_file, str):
             raise TypeError("Path to requirements file should be a string")
 
         with TempDir() as tmp_requirements_dir:
-            saved_requirements_dir_subpath = "requirements"
             _download_artifact_from_uri(
                 artifact_uri=requirements_file, output_path=tmp_requirements_dir.path()
             )
+            rel_path = posixpath.join(
+                _TORCHSERVE_ARTIFACTS_KEY, os.path.basename(requirements_file),
+            )
+            torchserve_artifacts_config[_TORCHSERVE_REQUIREMENTS_FILE_KEY] = {
+                "path": rel_path,
+                "uri": requirements_file,
+            }
             shutil.move(
-                tmp_requirements_dir.path(), posixpath.join(path, saved_requirements_dir_subpath)
+                tmp_requirements_dir.path(), posixpath.join(path, _TORCHSERVE_ARTIFACTS_KEY)
             )
 
     if extra_files:
+        torchserve_artifacts_config[_TORCHSERVE_EXTRA_FILES_KEY] = []
         if not isinstance(extra_files, list):
             raise TypeError("Extra files argument should be a list")
 
         with TempDir() as tmp_extra_files_dir:
-            saved_extra_files_dir_subpath = "artifacts"
             for extra_file in extra_files:
                 _download_artifact_from_uri(
                     artifact_uri=extra_file, output_path=tmp_extra_files_dir.path()
                 )
+                rel_path = posixpath.join(
+                    _TORCHSERVE_ARTIFACTS_KEY,
+                    _TORCHSERVE_EXTRA_FILES_KEY,
+                    os.path.basename(extra_file),
+                )
+                torchserve_artifacts_config[_TORCHSERVE_EXTRA_FILES_KEY].append(
+                    {"path": rel_path, "uri": extra_file}
```

Comment:
@harupy what's the intended use case for the URI here? I could imagine one convenience would be to be able to reference e.g. S3 URIs from the TorchServe plugin (for large `extra_files` like embeddings) rather than downloading them + reuploading them as an artifact for the current run. However, since we currently download the files locally anyways & include them alongside the model, maybe we could just expose the relative artifact path for now in `path`

Given that we always download the extra_files

Reply:
`mllfow.pyfunc.log_model` uses this format when saving the `artifacts` argument in `MLmodel`. I'm not sure what the original intention of this field is, but one benefit I came up with is we can tell where the file came from.

```
# Which foo.txt is this?
path: foo.txt
```

```
# foo.txt in my_bucket
path: foo.txt
uri: s3://my_bucket/foo.txt
```

Reply:
URI might contain confidential information though